### DISCUSSION TITLE
チェックするNG語

### DISCUSSION THREAD

**Discussion by noborus** [pgsql-jp/jpug-doc#2436] [NGとする語]
日本語ドキュメントに入ってはいけないNG語を決める

主に用語を統一するため、一方に統一したら他の用語をNGにしてチェックでエラーにします。

↓チェックするに至った背景
### 重要度

中

### 問題点

用語の統一に向けて、方針を決めて直したとしても新しく翻訳したときに長音付けルール等は見逃されて、
統一されない状態になってしまう。
そのため、GitHub Actions（と出来れば手元）でチェックして、指摘できると良い。

### 背景

#2395 #2280 で統一するように修正したら、手戻りがないようにしたい。

### 解決方法

NG語なリストを使用してチェックする。
方法を考えましたが、とりあえずは`git grep`で引っかかったらテスト失敗にする簡単な方法を入れるのが良いと思いました。

以下のようなファイルを置いて
ng.list 
```
。\s\+$
&#045;
からから
がが
をを
るる
キャスタ
子文字
少数点
ププランナ
デーや量
著者
ヘッダーファイル
### オーバ[^ー]
### ワーカ[^ー]
### スーパ[^ー]
```

※ \#\#\# はコメントではありませんが、まだわざと引っかからないようにしています。
※ 。の後の改行前のスペースもチェック。

`git grep -f ng.list doc/src/sgml/` で何も出力されないようにします。
Makefileはこんな感じで make checkしたときにチェックする。
```diff
diff --git a/doc/src/sgml/Makefile b/doc/src/sgml/Makefile
index da70668720c..687a4a73947 100644
--- a/doc/src/sgml/Makefile
+++ b/doc/src/sgml/Makefile
@@ -220,7 +220,7 @@ MAKEINFO = makeinfo
 ##
 
 # Quick syntax check without style processing
-check: postgres.sgml $(ALLSGML) check-tabs
+check: postgres.sgml $(ALLSGML) check-tabs check-ng
 	$(XMLLINT) $(XMLINCLUDE) --noout --valid $<
 
 
@@ -285,6 +285,17 @@ endif # sqlmansectnum != 7
 check-tabs:
 	@( ! grep '	' $(wildcard $(srcdir)/*.sgml $(srcdir)/ref/*.sgml $(srcdir)/*.xsl) ) || (echo "Tabs appear in SGML/XML files" 1>&2;  exit 1)
 
+# words to avoid as pgsql-jp
+check-ng:
+	@if [ -e $(top_srcdir)/taiyaku/ng.list ]; \
+	then \
+	  if git --no-pager grep -f taiyaku/ng.list *sgml ref/*sgml; \
+	  then \
+	    echo "使用NGな語が見つかりました。";\
+	    false;\
+	  fi \
+	fi
+
 ##
 ## Clean
 ##
```

github actionsは`ng.list`をダウンロードして、make checkをテストに追加するだけです。
ng.listは別レポジトリが良いと思います（taiyaku を考えています）。

### 注意点

本格的なlintツールが使用できるなら、そちらの方が良いです。
ただ一般的なルールはまだ使用できないため、ほとんど独自ルールになると思ってます。

### 貢献者として記載可否

以前の記載名を使用

**Comment by koizumistr** [comment]
てにをはについては

- を[をがてに]
- が[がをて]
- に[にがを]

はどうでしょうか。
あと、句読点で

- 。[。、]
- 、[。,]
- ,[。、]

もあってもいいかと。

**Comment by noborus** [comment]
ありがとうございます。

> てにをはについては
> 
> * を[をがてに]
> * が[がをて]
> * に[にがを]

現在は無いので、可能性は低いと思いますが、ひっかかる言葉が入ってくるかもしれません。「にがい」とか。

そうなると、別レポジトリにあるファイルが原因でgithub actionsでエラーになっていると修正が大変なので、
別レポジトリではなくjpug-docの中にNGリストを入れてしまう方が良いかもしれません。
チェックのほうを修正可能にしておけば、ほぼ使わない語が入ってしまっても良くなります。「内臓とか」。

> はどうでしょうか。 あと、句読点で
> 
> * 。[。、]
> * 、[。,]
> * ,[。、]
> 
> もあってもいいかと。

良いですね。

**Comment by noborus** [comment]
今までコンフリクトが発生するのを避けて、PRが出ているファイルは他の変更を入れないようにしていましたが、
用語の変更等で解決が自明の場合は、手戻り防止ができるので送っても良いように良いようにしたいです。

ということで例として一つ送ってみようと思います。

**Comment by noborus** [comment]
NGな語リストを検討するためにGitHub Discussionsを使用するのはどうでしょうか？
現状のリストはNGな語だけで、正しい語の表示等もできません。
そして追加するときにissueやpull requestに理由を書いて送ってもクローズされてしまえば、埋もれてしまって
追加した理由も探しづらくなります。

GitHub Discussions でNGな語に追加する提案から残しておくのが良いと思います。

**Comment by koizumistr** [comment]
> NGな語リストを検討するためにGitHub Discussionsを使用するのはどうでしょうか？

悪くない気はしますが、そもそも私がGitHub Discussionsを分かっていないのでした。Pull requestsの隣にあるタブ？ですよね。どこかに良い実例はないでしょうか。

**Comment by noborus** [comment]
実際に見てみないとわからないと思うのでやってみようと思います。

**Reply by koizumistr** [reply]
今ちょうど見ていたので、びっくりしました。こんなこと（ disccusion への移動）ができるのですね。

とReplyするてすと。

**Reply by koizumistr** [reply]
大きな項目があって、コメントが複数ついて、各コメントが下に伸びていく感じですね。スラドみたいな感じ？

**Reply by koizumistr** [reply]
カテゴリー、conversation、コメント って粒度があるようなので、どう使い分けるかですね。

**Comment by koizumistr** [comment]
この項目に追記していくのも一つの方法ですが、（「NGとする語」というカテゴリーを作って）1項目1NG語とする方法もあるかなと思いました。コメントには賛同を表す？上矢印があるので、投票に使えるかな、と。

カテゴリー内で項目が伸びていくか、項目内で伸びていくかの違いだけで、読みやすさはあまり変わらない、という可能性もありますが。

**Reply by koizumistr** [reply]
投票に上矢印を使うには「1項目1NG語」としないといけないような気がしていましたが、そんなことはないですね。決めておきさえすれば、今のままでもいけそう、、、

**Reply by noborus** [reply]
なるほど。カテゴリーごとに検索が効きそうなので、１語1項目でもよさそうです。

**Reply by noborus** [reply]
ちょっといろいろ試してみます

**Reply by koizumistr** [reply]
あと、今思いついたのですが、「〇〇の訳がぶれています」とissueで上げてくれる親切な人がいたとして、そのままdiscussionに持ってくるのなら、１語1項目もありなのかな、と思いました。

**Reply by noborus** [reply]
issueからの変換を使用して、項目を作ってみました。1項目1語のほうが良さそうです。

**Reply by koizumistr** [reply]
「何もしないでissueからの変換」で行けるかと思ったのですが、そうではないのですね。どう揃えたのかを追記しないと微妙な項目名に、、、、
ただ、どう揃えたのかを追記しておくと、カテゴリーを選択した場合の一覧性はとってもいいですね。一覧性と手間の天秤でしょうか。

**Reply by noborus** [reply]
はい。変えるのは後でも良いので、とりあえず変換だけでも良いと思います。
増えてきたらNGとなった語を検索でフィルタすると思うので、タイトルに ☓NG語→○OK語 的な表記がいいですかね...？

**Reply by koizumistr** [reply]
> タイトルに ☓NG語→○OK語 的な表記がいいですかね...？

いいですね（と脊髄反射的に思いました）。NG語とOK語の両方がパッと見て分かりますので。

**Reply by noborus** [reply]
変更してみました。

**Comment by noborus** [comment]
☓`&#045;`→○`&#45;`
☓からから→○から（連続しない）
☓を[をがてに]→○を（続けて書かない）
☓が[がをて]→○が（続けて書かない）
☓に[にがを]→○に（続けて書かない）
☓。\s\+$→○。（最後にスペースを入れない）
☓。[。、]→○。（続けて書かない）
☓、[。,]→○、（続けて書かない）
☓,[。、]→○,（続けて書かない）
☓キャスタ→○キャスト
☓子文字→○小文字
☓少数点→○小数点
☓内臓→○内蔵
☓ププランナ→○プランナ
☓デーや量→○データ量
☓外部結語→○外部結合
☓パーテイション→○パーティション
☓セマンティックス→○セマンティクス
☓セマンテックス→○セマンティクス
☓スタントアロン→○スタンドアロン
☓フェールオーバ→○フェイルオーバー
☓クライント→○クライアント
☓ロジカルプリケーション→○ロジカルレプリケーション
☓ロフ形式→○ログ形式
☓ライムライン→○タイムライン
☓メージャー→○メジャー
☓ケルベルス→○ケルベロス
☓アイデンティ[^テ]→○アイデンティティ（アイデンティではない）
☓著者→○作者
☓スタンドアロン→○スタンドアローン
☓エクゼキュータ→○エグゼキュータ
